sympy/geometry/ellipse.py

```diff
@@ -433,6 +455,69 @@ def equation(self, x='x', y='y'):
         t2 = ((y - self.center.y) / self.vradius)**2
         return t1 + t2 - 1
 
+    def equation_using_slope(self, slope):
+        """
+        Returns the equation of an ellipse using the slope of major axis, along with hradius
+        and vradius.
+
+        Approach to the solution :
+
+        Let the center of the ellipse be at C=(xc,yc)
+        Let the major axis be the line that passes through C with a slope of s;
+        points on that line are given by the zeros of L(x,y) = y−yc − s*(x−xc).
+        Let the minor axis be the line perpendicular to L (and also passing through C);
+        points on that line are given by the zeros of l(x,y) = s*(y−yc) + (x−xc).
+
+        The ellipse is then defined by the zeros of
+        E(x,y) = L(x,y)**2/a + l(x,y)**2/b − 1
+
+        Requiring that the distance between the intersections of E and L be 2M identifies
+        b = M**2 * ( 1 + s**2 )
+
+        and similarly, requiring that the intersections between E and l be separated by 2m identifies
+        a = m**2 * ( 1 + s**2 )
+
+        So the general equation of the ellipse centered at (xc,yc) whose major axis (with radius of M)
+        is on a line with slope s, and whose minor axis has radius of m, is given by the solutions of:
+
+        ( ( y − yc ) − s ( x − xc ) )**2         ( s * ( y − yc ) + ( x − xc ) )**2
+        --------------------------------   +     ----------------------------------   =  1
+                m**2 * ( 1 + s**2 )                  M**2 * ( 1 + s**2 )
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Ellipse
+        >>> Ellipse(Point(1,0), 4, 2).equation_using_slope(1)
+        (-x + y + 1)**2/8 + (x + y - 1)**2/32 - 1
+
+        Reference
+        =========
+
+        https://math.stackexchange.com/questions/108270/what-is-the-equation-of-an-ellipse-hat-is-not-aligned-with-the-axis/646971
+        """
+
+        a, b, x, y, m, M, x_c, y_c, s = var('a,b,x,y,m,M,x_c,y_c,s')
```

Comment:
don't use `var`, but `symbols`. `var` injects the symbols into the parent scope.

Reply:
More over, the user should be able to set these symbols, so that he can later reference them. Probably passing them as arguments, optionally?

Reply:
@avishrivastava11, I agree with what @asmeurer said.